src/boilerplate/common/bin/setup

```diff
@@ -5,6 +5,74 @@ GREEN='\033[0;32m'
 RED='\033[0;31m'
 NC='\033[0m'
 
+while getopts "n:a:m:k:r:" arg; do
+  case $arg in
+    n)
+      network=$OPTARG
+      echo networkvalue $OPTARG 
+      ;;
+    a)
+      account=$OPTARG
+      echo accountvalue $OPTARG
+      ;;
+    m)
+      mnemonic=$OPTARG
+      echo mnemonicvalue $OPTARG
+      ;;
+    k)
+      key=$OPTARG
+      echo keyvalue $OPTARG
+      ;;
+    r)
+      rpc=$OPTARG
+      echo rpcvalue $OPTARG
+      ;;
+  esac
+done
+
+cp docker-compose.zapp.override.default.yml docker-compose.zapp.override.yml
+
+cp entrypoint_default.sh entrypoint.sh
+
+cp bin/default_startup bin/startup
+
+perl -i -pe "s,docker-compose.zapp.yml -f docker-compose.zapp.override.yml,docker-compose.zapp.yml,g" package.json
+
+if [[ $network == 'polygon' ]] || [[ $network == 'sepolia' ]] || [[ $network == 'goerli' ]]
+then
+perl -i -pe "s,DEFAULT_ACCOUNT: '',DEFAULT_ACCOUNT: \'$account\',g" docker-compose.zapp.override.yml
+perl -i -pe "s,DEFAULT_ACCOUNT_MNEMONIC: '',DEFAULT_ACCOUNT_MNEMONIC: \'$mnemonic\',g" docker-compose.zapp.override.yml
+perl -i -pe "s,KEY: '',KEY: \'$key\',g" docker-compose.zapp.override.yml
+perl -i -pe "s,docker-compose.zapp.yml up,docker-compose.zapp.yml -f docker-compose.zapp.override.yml up,g" bin/startup
+perl -i -pe "s,docker-compose.zapp.yml,docker-compose.zapp.yml -f docker-compose.zapp.override.yml,g" package.json
+perl -i -pe "s,docker-compose -f docker-compose.zapp.yml up -d ganache, ## up ganache service for ganache,g" bin/startup
+fi
+
+if [[ $network == 'polygon' ]]
+then
+perl -i -pe "s,RPC_URL: '',RPC_URL: \'wss:\/\/rpc-mumbai.maticvigil.com\/ws\/v1\/$rpc\',g" docker-compose.zapp.override.yml
+perl -i -pe "s,development_ganache,polygon,g" entrypoint.sh
+perl -i -pe "s,while,##while,g" entrypoint.sh
+perl -i -pe "s,10,60,g" bin/startup
+fi
+
+
+if [[ $network == 'sepolia' ]]
+then
+perl -i -pe "s,RPC_URL: '',RPC_URL: \'wss://sepolia.infura.io/ws/v3//$rpc\',g" docker-compose.zapp.override.yml
+perl -i -pe "s,development_ganache,sepolia,g" entrypoint.sh
+perl -i -pe "s,while,##while,g" entrypoint.sh
+perl -i -pe "s,10,500,g" bin/startup
+fi
+
+if [[ $network == 'goerli' ]]
+then
+perl -i -pe "s,RPC_URL: '',RPC_URL: \'wss://goerli.infura.io/ws/v3/$rpc\',g" docker-compose.zapp.override.yml
+perl -i -pe "s,development_ganache,goerli,g" entrypoint.sh
+perl -i -pe "s,while,##while,g" entrypoint.sh
+perl -i -pe "s,10,500,g" bin/startup
+fi
+
```

Comment:
This is really smart work!
I have a couple of small comments/suggestions:
- Could we separate out code to initialise the deployment and the zkp setup? I am thinking of a case where a user may want to deploy on a different network or change keys, but doesn't need to re-do trusted setup.
- Could we rename `polygon` to `mumbai`? Just to keep consistent with other testnet names here :) 

For the first one, we could either have two `bin` files, one for `setup` and one something like `deploy-config`, or have one file with an option to complete the trusted setup. I'm not sure of the best way of the two, happy to discuss

Reply:
- Seperated setup and deployment configuration with a flag `-s`. If no setup is required , do `./bin/setup -s n`. Also setup is forced , if `proving-files` directory is missing.
- Renamed `polygon` to `mumbai`